manifests/params.pp

```diff
@@ -168,19 +168,21 @@
   $auth_allowed = ['$1']
 
   # Will this host be a puppet agent ?
-  $agent                     = true
-  $remove_lock               = true
-  $client_certname           = $::clientcert
+  $agent                      = true
+  $remove_lock                = true
+  $client_certname            = $::clientcert
 
   # Custom puppetmaster
-  if defined('$trusted') and $::trusted['authenticated'] == 'local' {
-    $puppetmaster            = undef
+  # needed due to a PUP-4072
+  # more information in https://github.com/theforeman/puppet-foreman/commit/5fe3239da0c6fbac76172f61042a69ab3a7eb4e6
+  if versioncmp($::puppetversion, '3.7.5') < 0 or defined('$::puppetmaster') {
```

Comment:
Doesn't this change the behavior? IIRC this was needed in case of masterless setups but I don't have such a setup.